first homework done

Kouler opened the pull request.

Kouler committed:
first homework done

Kouler committed:
codestyle fixed

vasilmik commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,23 @@
+
+    private static final int SECONDS = 60;
+
+    private Task1() {
```
Удалите если не используете

Kouler replied:
немного не понял кого удалить... Пустой конструктор? Так у меня тогда стили ругаются

Kouler replied:
> Удалите если не используете

немного не понял кого удалить... Пустой конструктор? Так у меня тогда стили ругаются

vasilmik commented on src/main/java/edu/hw1/Task1.java:
```diff
@@ -0,0 +1,23 @@
+    public static long minutesToSeconds(String s) {
+        String[] parsedS = s.split(":");
+        long min = Long.parseLong(parsedS[0]);
+        short sec = Short.parseShort(parsedS[1]);
```
Тут может быть ошибка

vasilmik commented on src/main/java/edu/hw1/Task3.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    public static boolean isNestable(int[] a1, int[] a2) {
+        int minA1 = a1[0];
```
можно проверить массив

vasilmik commented on src/main/java/edu/hw1/Task6.java:
```diff
@@ -0,0 +1,59 @@
+
+        // Диапазон значений для нашего числа как массива не большой - всего 10 значений,
+        // а => будем сортировать подсчётом за O(n)
+        for (; x != KAPREKAR; count++) { // решил так, не хочу в while писать count++ снизу
```
Так писать не нужно, обычно переменные внутри for не должны выходить за границу видимости

vasilmik commented on src/main/java/edu/hw1/Task6.java:
```diff
@@ -0,0 +1,59 @@
+        for (; x != KAPREKAR; count++) { // решил так, не хочу в while писать count++ снизу
+            //массив подсчёта
+            int[] countArr = new int[TEN];
+            for (int i = 0; i < KAPLENGTH; i++) {
```
Попробуйте зарефакторить код чтобы его можно было прочитать В Т есть 3 действия: 
1. Расположим цифры сначала в порядке возрастания
2. затем в порядке убывания.
3. Вычтем из большего меньшее. Производя перестановки цифр и вычитания, нули следует сохранять.

плюс нужно сделать рекурсивно

Kouler replied:
> Попробуйте зарефакторить код ...
То есть мне нужно написать код, который будет делать именно вот прям пошагово так, как написано в методе, как бы решал человек?
Не совсем понял что именно зарефакторить надо, т.к. у меня, вроде, программа сначала обнаруживает, какие цифры используются, затем собирает два числа из них, одно по неубыванию, второе по невозрастанию и затем так же находит следующее число для итерации, пока не обнаружим Капрекара.
При этом мне не нужно учитывать такое понятие, как "сохранение нулей", т.к. они автоматом математикой убираются

vasilmik commented on src/test/java/edu/hw1/Task7Test.java:
```diff
@@ -0,0 +1,54 @@
+    @Test
+    @DisplayName("Task7: binary rotate left")
+    void check_rotateLeft() {
+        long[][] set = {
```
Проверьте еще раз
rotateRight(8, 1) -> 4 // 1000 -> 0100 
rotateLeft(16, 1) -> 1 // 10000 -> 00001 
rotateLeft(17, 2) -> 6 // 10001 -> 00110

Kouler replied:
> Проверьте еще раз 
Немного не понял, что проверить.
По тестам rotateRight верно возвращает 4 при введённом (8, 1), а rotateLeft возвращает 1 и 6 при введённых (16, 1) и (17, 2) соответственно, тесты проходят успешно
![image](https://github.com/Kouler/Tinkoff/assets/44964545/5fd9583c-e6dd-4cb1-b56a-9758136099a9)
![image](https://github.com/Kouler/Tinkoff/assets/44964545/9688bd9d-b59a-4932-8487-5581acd9ca6f)
![image](https://github.com/Kouler/Tinkoff/assets/44964545/a4f08ec4-b95f-44d2-a98e-37af1cff1211)

Kouler committed:
Task1 - Added string checks for only one ':' and only numbers;
Task1Test - Added a couple of tests for string checking;
Task3 - Added checking the arrays;
Task6 - For cycle changed to while.

Kouler committed:
corrected a little mistake in test6

Kouler commented:
Я исправил, что получилось понять, по рефактору кода не совсем понял, что именно не так, +  не понял в чем проблема 7 таска (или теста), жду, когда ответите)
